infrastructure/persistence/file.go

```
@@ -159,34 +163,113 @@
+	if err != nil {
+		closer()
+		return nil, nil, err
+	}
+	// xzReader doesn't require a close; use original closer.
+	return xzReader, closer, nil
+}
+
+func wrapZstdReader(reader io.Reader, closer func() error) (io.Reader, func() error, error) {
+	decoder, err := zstd.NewReader(reader)
+	if err != nil {
+		closer()
```

golangci-lint (1.64.8)

Error return value is not checked

(errcheck)